File: RoomMate.Front/RoomMatePortal/src/app/user-controll-panel/register/dto/register-dto.ts
import { RolesEnum } from "../../dto/RolesEnum";
import { AddressFlatDto } from "src/app/address/dto/address-dto";

export class RegisterDto {
  public login: string;
  public name: string;
  public surname: string;
  public password: string;
  public email: string;
  public addressDto: AddressFlatDto = new AddressFlatDto();
}

File: RoomMate.Domain/Dto/RegisterDto.cs
namespace RoomMate.Domain.Dto
{
    public class RegisterDto : BaseDto
    {
        public string Login { get; set; }
        
        public string Name { get; set; }

        public string Surname { get; set; }

        public string Password { get; set; }

        public string Email { get; set; }

        public string Code { get; set; }

        public string PasswordSalt { get; set; }


        public AddressFlatDto AddressDto { get; set; }


        public bool IsValid()
        {
            return (IsEmpty(this.Login) && IsEmpty(this.Name)
                && IsEmpty(this.Surname)  && ValidationDto.IsValidPassword(this.Password)
                && ValidationDto.IsValidEmail(this.Email) && IsValidId(this.AddressDto.CityId));
        }
    }
}
